src/MongoDB.Driver.Core/Core/Servers/ServerMonitor.cs

```diff
@@ -93,7 +93,7 @@ public void Initialize()
 
         public void Invalidate()
         {
-            SetDescriptionIfChanged(_baseDescription);
+            SetDescription(_baseDescription.With(lastUpdateTimestamp: DateTime.UtcNow));
```

Comment:
Do you need to update the timestamp on line 235 also?

It's tricky... because I think that even though line 235 does NOT update the timestamp, the only way to reach line 235 is if there was a heartbeat exception, in which case we will also reach line 240, which DOES update the timestamp.

So it's probably working, but when I was searching all uses of _baseDescription to make sure we never used it without updating the timestamp line 235 initially looked like a bug.

Maybe better to just be explicit about it?

What do you think? If you are happy with the current code I am willing to LGTM as is.

Reply:
I like the explicit updating of the timestamp: updated!